File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int maxi=prices[prices.size()-1];
        int ans=0;
        for(int i=prices.size()-2;i>=0;i--) {
            if(prices[i]<=maxi) {
                ans=max(ans,maxi-prices[i]);
            }
            maxi=max(maxi,prices[i]);
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxi=0,mini=prices[0];
        for(int i=1;i<prices.length;i++) {
            maxi = Math.max(maxi,prices[i]-mini);
            mini = Math.min(mini,prices[i]);
        }
        return maxi;
    }
}